samples/sqltransport/multi-instance/version_3/EndpointConnectionLookup/ConnectionProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+
+namespace EndpointConnectionStringLookup
+{
+    #region ConnectionProvider
+    public class ConnectionProvider
+    {
+        private const string ReceiverConnectionString = @"Data Source=.\SQLEXPRESS;Initial Catalog=ReceiverCatalog;Integrated Security=True";
+        private const string SenderConnectionString = @"Data Source=.\SQLEXPRESS;Initial Catalog=SenderCatalog;Integrated Security=True";
+
+        public static async Task<SqlConnection> GetConnecton(string transportAddress)
+        {
+            var connectionString = transportAddress.Contains("MultiInstanceReceiver") ? ReceiverConnectionString : SenderConnectionString;
```

Comment:
It would be better to include here the full transport address.
Looking at it now I think we also should describe transport address structure in the configuration page, where we talk about mutli-instance setup.